### PR TITLE
Update rox-ci-image

### PR DESCRIPTION
**Pull request by roxbot**
Hello,
This is an automated PR created to bump the base image.
It was created from https://github.com/stackrox/rox-ci-image/pull/107.

This PR also moves all test image dependencies quay.io.

### CODE REVIEW

**Comment by janisz** [comment]
https://issues.redhat.com/browse/ROX-8114

**Review by janisz — APPROVED** [review]

File: functionaltest-jenkins-plugin/build.gradle [hunk 1]
```diff
@@ -70,4 +70,6 @@ dependencies {
     implementation 'io.gsonfire:gson-fire:1.8.5'
     implementation 'org.openapitools:jackson-databind-nullable:0.2.2'
     implementation "jakarta.annotation:jakarta.annotation-api:2.0.0"
+    implementation 'javax.annotation:javax.annotation-api:1.3.2'
+    implementation 'javax.xml.bind:jaxb-api:2.3.1'
```
Comment (janisz) [inline functionaltest-jenkins-plugin/build.gradle:74]: Why do we need that?

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,13 +1,19 @@
 version: 2.1
+
+defaultImage: &defaultImage
+  image: "quay.io/rhacs-eng/apollo-ci:snapshot-jenkins-plugin-0.3.26-19-g230e4fc30a" # TODO(do not merge): After upstream PR is merged, cut a tag and update this
```
Comment (janisz) [inline .circleci/config.yml:4]: What PR needs to be merged?

**Review by janisz — CHANGES_REQUESTED** [review]

File: functionaltest-jenkins-plugin/build.gradle (hunk 1 above)
Comment (gavin-stackrox) [reply functionaltest-jenkins-plugin/build.gradle:74]: https://app.circleci.com/pipelines/github/stackrox/jenkins-plugin/1107/workflows/46558806-ba47-45f7-9146-1afba31ec939/jobs/3002
&
https://app.circleci.com/pipelines/github/stackrox/jenkins-plugin/1106/workflows/63ee3fa3-5ef4-4a4b-b831-5bb353790335/jobs/2999

File: .circleci/config.yml (hunk 2 above)
Comment (gavin-stackrox) [reply .circleci/config.yml:4]: https://github.com/stackrox/rox-ci-image/pull/107

File: functionaltest-jenkins-plugin/build.gradle (hunk 1 above)
Comment (janisz) [reply functionaltest-jenkins-plugin/build.gradle:74]: OK, so I guess it's related to JDK version used in container

**Review by janisz — APPROVED** [review]

**Pull request merged**